posthog/models/feature_flag/flag_matching.py

```diff
@@ -511,174 +513,173 @@ def variant_lookup_table(self, feature_flag: FeatureFlag):
     @cached_property
     def query_conditions(self) -> dict[str, bool]:
         try:
-            # Some extra wiggle room here for timeouts because this depends on the number of flags as well,
-            # and not just the database query.
-            all_conditions: dict = {}
-            person_query: QuerySet = Person.objects.db_manager(DATABASE_FOR_PERSONS).filter(
-                team_id=self.team_id,
-                persondistinctid__distinct_id=self.distinct_id,
-                persondistinctid__team_id=self.team_id,
-            )
-            basic_group_query: QuerySet = Group.objects.db_manager(DATABASE_FOR_FLAG_MATCHING).filter(
-                team_id=self.team_id
-            )
-            group_query_per_group_type_mapping: dict[GroupTypeIndex, tuple[QuerySet, list[str]]] = {}
-            # :TRICKY: Create a queryset for each group type that uniquely identifies a group, based on the groups passed in.
-            # If no groups for a group type are passed in, we can skip querying for that group type,
-            # since the result will always be `false`.
-            for group_type, group_key in self.groups.items():
-                group_type_index = self.cache.group_types_to_indexes.get(group_type)
-                if group_type_index is not None:
-                    # a tuple of querySet and field names
-                    group_query_per_group_type_mapping[group_type_index] = (
-                        basic_group_query.filter(group_type_index=group_type_index, group_key=group_key),
-                        [],
-                    )
-
-            person_fields: list[str] = []
-
-            for existence_condition_key in self.has_pure_is_not_conditions:
-                if existence_condition_key == PERSON_KEY:
-                    person_exists = person_query.exists()
-                    all_conditions[f"{ENTITY_EXISTS_PREFIX}{PERSON_KEY}"] = person_exists
-                else:
-                    if existence_condition_key not in group_query_per_group_type_mapping:
-                        continue
-
-                    group_query, _ = group_query_per_group_type_mapping[cast(GroupTypeIndex, existence_condition_key)]
-                    group_exists = group_query.exists()
-                    all_conditions[f"{ENTITY_EXISTS_PREFIX}{existence_condition_key}"] = group_exists
-
-            def condition_eval(key, condition):
-                expr = None
-                annotate_query = True
-                nonlocal person_query
-
-                property_list = Filter(data=condition).property_groups.flat
-                properties_with_math_operators = get_all_properties_with_math_operators(
-                    property_list, self.cohorts_cache, self.project_id
+            with execute_with_timeout(FLAG_MATCHING_QUERY_TIMEOUT_MS * 2, READ_ONLY_DATABASE_FOR_PERSONS):
```

Comment:
This looks like a big change but it's basically reverting https://github.com/PostHog/posthog/pull/31055/files#diff-dd0c8b02e1bb38052f9d4a3cb8315b2750639ffc645f1158c1520e4df482596bR516

Reply:
yeah, looks good.